src/lazy/binary/raw/v1_1/immutable_buffer.rs

```diff
@@ -728,68 +736,137 @@ impl<'a> ImmutableBuffer<'a> {
         opcode: Opcode,
     ) -> ParseResult<'a, EncodedAnnotations> {
         let input_after_opcode = self.consume(1);
-        // TODO: This implementation actively reads the annotations, which isn't necessary.
-        //       At this phase of parsing we can just identify the buffer slice that contains
-        //       the annotations and remember their encoding; later on, the annotations iterator
-        //       can actually do the reading. That optimization would be impactful for FlexSyms
-        //       that represent inline text.
-        let (sequence, remaining_input) = match opcode.low_nibble() {
+        let (header_length, remaining_input) = match opcode.low_nibble() {
             7 => {
-                let (flex_sym, remaining_input) = input_after_opcode.read_flex_sym()?;
-                let sequence = EncodedAnnotations {
-                    encoding: AnnotationsEncoding::FlexSym,
-                    header_length: 1, // 0xE7
-                    sequence_length: u16::try_from(flex_sym.size_in_bytes()).map_err(|_| {
-                        IonError::decoding_error(
-                            "the maximum supported annotations sequence length is 65KB.",
-                        )
-                    })?,
-                };
```

Comment:
🗺️ Previously I was constructing an `EncodedAnnotations` struct in each `match` arm. I lifted out out to the end of the method, which I think helped readability.